components/proxy/src/main/java/com/hotels/styx/routing/handlers/PathPrefixRouter.java

```diff
@@ -97,29 +112,25 @@ public RoutingObject build(List<String> fullName, Context context, StyxObjectDef
                 throw missingAttributeError(configBlock, join(".", fullName), "routes");
             }
 
-            PathPrefixRouter pathPrefixRouter = new PathPrefixRouter(
-                    config.routes.stream()
-                            .map(route -> pair(route.prefix, Builtins.build(ImmutableList.of(""), context, route.destination)))
-                            .collect(toList())
-            );
-
-            return new RoutingObject() {
-                @Override
-                public Eventual<LiveHttpResponse> handle(LiveHttpRequest request, HttpInterceptor.Context context) {
-                    return pathPrefixRouter.route(request)
-                            .orElse((x, y) -> Eventual.error(new NoServiceConfiguredException(request.path())))
-                            .handle(request, context);
-                }
-
-                @Override
-                public CompletableFuture<Void> stop() {
-                    pathPrefixRouter.routes.forEach((route, routingObject) -> routingObject.stop());
-                    return completedFuture(null);
-                }
-            };
+            List<PrefixRoute> routes = new ArrayList<>();
+            for (PathPrefixConfig routeConfig : config.routes()) {
+                String prefix = routeConfig.prefix();
+                RoutingObject route = Builtins.build(singletonList(""), context, routeConfig.destination());
+                routes.add(new PrefixRoute(prefix, route));
+            }
```

Comment:
Please keep using the functional stream/map based approach to build the `routes` list. Because that is a standard idiom used elsewhere in the Styx codebase, and I believe it is good to keep it consistent.

Reply:
Since the `routes` structure is purely internal I was thinking of replacing it with an array which would be easier for the CPU to iterate over in the main function of this class. I'll update the PR and we can discuss that implementation.

Reply:
In the idiomatic styx code you would do it something like this:

```
            ArrayList<PrefixRoute> x = config.routes().stream()
                    .map(route -> new PrefixRoute(
                            route.prefix(),
                            Builtins.build(singletonList(""), context, route.destination())))
                    .sorted()
                    .collect(toCollection(ArrayList::new));
```